### PR TITLE
[BD] Add tests with python 38.

### PR DESCRIPTION
**Pull request by morenol**
Add tests with python3.8 and allow failures in python 3.8 workers.
Note: python3.8 tests are failing as this is being writing due to python3.8 interpreter is not installed in 
docker container used in the tests.

Upgraded bok-choy and selenium.

## Reviewers
- [x] @awais786 
- [ ] @jmbowman 
- [x] Ready for edX review.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @morenol! I've created [OSPR-4394](https://openedx.atlassian.net/browse/OSPR-4394) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: requirements/local.txt [hunk 1]
```diff
@@ -70,7 +70,7 @@ pbr==5.4.5                # via -r requirements/test.txt, mock, stevedore
 pep257==0.7.0             # via -r requirements/test.txt
 pillow==7.1.1             # via -r requirements/test.txt, needle
 pinax-announcements==3.0.2  # via -r requirements/test.txt
-pip-tools==4.5.1          # via -r requirements/pip_tools.txt
+pip-tools==5.0.0          # via -r requirements/pip_tools.txt
```
Comment (jmbowman) [inline requirements/local.txt]: You may want to constrain this for now; 5.0.0 only works with pip 20+, and all our VMs and containers are still installing 19.3.1 by default.  I'm starting to work on getting that fixed.

File: tox.ini [hunk 2]
```diff
@@ -63,45 +68,6 @@ commands =
     waffle_learner_analytics: python manage.py waffle_flag enable_learner_analytics --create --everyone
     waffle_course_filters: python manage.py waffle_switch enable_course_filters on --create
     waffle_course_passing: python ./manage.py waffle_switch enable_course_passing on --create
+    accept: xvfb-run pytest -v acceptance_tests --ignore=acceptance_tests/course_validation
+    a11y: xvfb-run  pytest -v a11y_tests -k 'not NUM_PROCESSES=1' --ignore=acceptance_tests/course_validation
```
Comment (jmbowman) [inline tox.ini]: You shouldn't need xvfb-run if you run the browser in headless mode: https://bok-choy.readthedocs.io/en/latest/testing.html#testing-via-travisci .  May need to upgrade Firefox first if it's using a very old version.

File: requirements/constraints.txt [hunk 3]
```diff
@@ -52,6 +52,8 @@ gunicorn<20.0.0
 # 2.0.0 drops Python 2.7 support.
 mysqlclient<2.0.0
```
Comment (mzulqarnain1) [inline requirements/constraints.txt]: We should drop the constraints like this that are no longer needed.

**Review by mzulqarnain1 — APPROVED** [review]

File: requirements/constraints.txt (hunk 3 above)
Comment (morenol) [reply requirements/constraints.txt]: Done

File: requirements/constraints.txt [hunk 4]
```diff
@@ -16,44 +16,11 @@ Django<3.0.0
...
 
-# 3.0.0 drops Python 2.7 support
-edx-django-utils<3.0.0
-
-# gunicorn version 20.0.0 dropped support to Python 2.7
-gunicorn<20.0.0
-
-# 2.0.0 drops Python 2.7 support.
-mysqlclient<2.0.0
+# pip20+ requires for greater version of pip-tools. But installing 19.3.1 in all our containers and VMs.
```
Comment (awais786) [inline requirements/constraints.txt:19]: You don't need this condition anymore. You can check the comment [here](https://github.com/edx/xss-utils/pull/5#discussion_r415950830)

Comment (awais786) [reply requirements/constraints.txt:19]: I think it's ok because its not a library.

**Review by awais786 — APPROVED** [review]

**Review by awais786 — APPROVED** [review]

**Comment by morenol** [comment]
@awais786 This is ready

**Comment by openedx-webhooks** [comment]
@morenol 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-analytics-dashboard&entry.867055334=https://github.com/edx/edx-analytics-dashboard/pull/1007&entry.1484655318=https://github.com/morenol&entry.752974735=2020-04-16+18:29&entry.1917517419=2020-05-08+15:52&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**